dotCMS/src/main/java/com/dotcms/aspects/MethodInterceptor.java

```diff
@@ -0,0 +1,18 @@
+package com.dotcms.aspects;
+
+import java.io.Serializable;
+
+/**
+ * The interceptor handles the invoke to a method on an AOP context.
+ *
+ * @param <T>
+ */
+public interface MethodInterceptor<T> extends Serializable {
+
+    /**
+     * This method will be called when the matcher catch the invoke to the target method.
+     * @param delegate DelegateMethodInvocation
+     * @return Object object to return as part of the method invoke.
+     */
+    Object invoke (final DelegateMethodInvocation<T> delegate) throws Throwable;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)